### DISCUSSION TITLE
Creating rpm/dpkg packages for quay together with a bootstrapping script

### DISCUSSION THREAD

**Discussion by brhenc** [quay/quay#3448] [General]
Greetings,

I was thinking of creating rpm packages for quay to make it easier to get started with quay for system administrators and network engineers.

Most container registries (for example, harbor) require a pretty involved deployment on kubernetes.
Zot (https://zotregistry.dev/v2.1.0/install-guides/install-guide-linux/) only provides binaries.

Having a package in the main package repositories (or as a COPR/Ubuntu PPA) lowers the barrier to entry significantly.
Together with a bootstrapping script (shipped as part of the package) that bootstraps quay, it should be possible to simplify the install down to "dnf -y install quay && /usr/local/bin/quay-bootstrap.sh". 

I am skilled at packaging and should be able to deal with the dependencies (and handling python dependency issues).

The notable example of how useful packaging up-to-date packages is ansible, which has a ubuntu ppa and enjoys wide
adoption across the different linux distributions.

Thank you in advance for any feedback on the matter - I think it might be worthwhile to pursue but if it creates a burden on the quay team (bug reports due to python dependency issues) it might be best to consider an alternative packaging method (like snap.io packages) or sticking to container images only.

**Comment by ibazulic** [comment]
Hey there,

Quay is distributed as a container image. Images can be found under:

~~~
quay.io/projectquay/quay
~~~

So you only need a container engine runtime to run Quay (along with some storage capabilities and a database). It should be fairly easy to deploy it on any platform that satisfies those conditions. Quay was never intended to be built as an RPM or DEB package. One of the reasons is a fairly big set of Python/JS dependencies that Quay relies on.

Hope that helps!
